src/main/java/service/impl/FruitTransactionParserImpl.java

```diff
@@ -0,0 +1,20 @@
+package service.impl;
+
+import model.FruitTransaction;
+import service.FruitTransactionParser;
+
+public class FruitTransactionParserImpl implements FruitTransactionParser {
+    @Override
+    public FruitTransaction parse(String line) {
+        String[] parts = line.split(",");
+        FruitTransaction.Operation operation = FruitTransaction.Operation.fromCode(parts[0]);
+        String fruit = parts[1];
+        int quantity = Integer.parseInt(parts[2]);
+
+        FruitTransaction transaction = new FruitTransaction();

```

Comment:
Instead of setting properties individually after object creation, consider creating a constructor in `FruitTransaction` that takes all the required properties as parameters.